src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/domainevents/DomainEventDescriber.kt

```diff
@@ -121,9 +123,22 @@ class DomainEventDescriber(
     return event.describe { "A placement was not made for the placement request.$failureReason" }
   }
 
+  @SuppressWarnings("TooGenericExceptionThrown")
   private fun buildBookingCancelledDescription(domainEventSummary: DomainEventSummary): String? {
     val event = domainEventService.getBookingCancelledEvent(domainEventSummary.id())
-    return event.describe { "The placement was cancelled. The reason was: '${it.eventDetails.cancellationReason}'" }
+    return event.describe { data ->
+      val booking = bookingRepository.findByIdOrNull(data.eventDetails.bookingId)
+        ?: throw RuntimeException("Booking ID ${data.eventDetails.bookingId} with cancellation not found")
+      if (booking.cancellations.count() != 1) {
+        throw RuntimeException("Booking ID ${data.eventDetails.bookingId} does not have one cancellation")
+      }
+      val cancellation = booking.cancellations.first()
```

Comment:
we should be using the cancellation id that added as metadata via https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/2038

But - we'll need to start adding the metadata to the DomainEventSummary object to achieve this :( I think it may just be a case of adding `d.metadata` to the query in DomainEventRepository.findAllTimelineEventsByApplicationId and adding the map definition to the DomainEventSummary interface?

We can then traverse cancellation.booking to get hold of the booking. This should at least reduce the amount of potential errors we deal with here?

Reply:
Discussed with Rob. Getting metadata via the JPQL is proving tricky as this doesn't appear to be supported (loading in map).

We think the correct solution is to actually use the DomainEventEntity directly here, and add explicit relationships to linked entities (e.g. ApplicationEntity, BookingEntity etc.). We need to be careful these aren't loaded by default (eagerly) but in this query (or whatever we replace it with) we _do_ load these eagerly